controller/collaborators_blackbox_test.go

```diff
@@ -406,88 +445,116 @@ func (s *CollaboratorsControllerTestSuite) TestRemoveSingleCollaborator() {
+			space := g.CreateSpace().AddAdmin(admin).AddContributor(contrib)
+			spaceID, _ := uuid.FromString(space.SpaceID())
+			svc, ctrl := s.NewUnsecuredController()
 			// when/then
-			test.RemoveCollaboratorsUnauthorized(t, svc.Context, svc, ctrl, spaceID, s.testIdentity2.ID.String())
+			test.RemoveCollaboratorsUnauthorized(t, svc.Context, svc, ctrl, spaceID, contrib.IdentityID().String())
 		})
 
-		t.Run("deprovisionned user ", func(t *testing.T) {
+		t.Run("deprovisionned user account", func(t *testing.T) {
 			// given
-			spaceID := s.createSpace(t)
-			svc, ctrl := s.UnSecuredControllerDeprovisionedUser()
+			g := s.NewTestGraph(t)
+			admin := g.CreateUser()
+			contrib := g.CreateUser()
+			space := g.CreateSpace().AddAdmin(admin).AddContributor(contrib)
+			spaceID, _ := uuid.FromString(space.SpaceID())
+			svc, ctrl := s.NewUnsecuredControllerDeprovisionedUser()
 			// when/then
-			test.RemoveCollaboratorsUnauthorized(t, svc.Context, svc, ctrl, spaceID, s.testIdentity2.ID.String())
+			test.RemoveCollaboratorsUnauthorized(t, svc.Context, svc, ctrl, spaceID, contrib.IdentityID().String())
 		})
 	})
 
 	s.T().Run("forbidden", func(t *testing.T) {
 		// given
 		g := s.NewTestGraph(t)
-		ownerIdentity := g.CreateUser().Identity()
-		spaceID := s.createSpaceByIdentity(t, ownerIdentity)
-		toRemoveIdentity := g.CreateUser().Identity()
-		svc, ctrl := s.SecuredController()
+		admin := g.CreateUser()
+		space := g.CreateSpace().AddAdmin(admin)
+		spaceID, _ := uuid.FromString(space.SpaceID())
+		svc, ctrl := s.NewSecuredController(admin.Identity())
 		_, actualUsers := test.ListCollaboratorsOK(t, svc.Context, svc, ctrl, spaceID, nil, nil, nil, nil)
-		checkCollaborators(t, []uuid.UUID{ownerIdentity.ID}, actualUsers)
+		checkCollaborators(t, actualUsers, admin.Identity())
 		currentIdentity := g.CreateUser().Identity()
-		svc, ctrl = s.SecuredControllerForIdentity(currentIdentity)
+		svc, ctrl = s.NewSecuredController(currentIdentity)
 		// 403 from Auth
 		// We have to allow any OSIO user to list collaborators. See https://github.com/fabric8-services/fabric8-auth/pull/521 for details
 		//test.ListCollaboratorsForbidden(t, svc.Context, svc, ctrl, spaceID, nil, nil, nil, nil)
 		test.ListCollaboratorsOK(t, svc.Context, svc, ctrl, spaceID, nil, nil, nil, nil)
 
-		// when
-		payload := &app.AddManyCollaboratorsPayload{Data: []*app.UpdateUserID{{ID: g.CreateUser().IdentityID().String(), Type: idnType}}}
-		test.AddCollaboratorsForbidden(t, svc.Context, svc, ctrl, spaceID, g.CreateUser().IdentityID().String())
-		test.AddManyCollaboratorsForbidden(t, svc.Context, svc, ctrl, spaceID, payload)
-		rPayload := &app.RemoveManyCollaboratorsPayload{Data: []*app.UpdateUserID{{ID: toRemoveIdentity.ID.String(), Type: idnType}}}
-		test.RemoveManyCollaboratorsForbidden(t, svc.Context, svc, ctrl, spaceID, rPayload)
+		t.Run("add", func(t *testing.T) {
+			// when
+			payload := newAddManyCollaboratorsPayload(t, g.CreateUser().Identity())
```

Comment:
No need to create a new TestGraph here?

Reply:
the thing is, the data is not removed after each subtest, so here, we create a data set and check in the 2 subsequent subtests that operations are forbidden. I reckon that we could do the same (less data init) on other tests as well

Reply:
This is where you start make even deeper not-explicit dependencies between subtests which will make maintenance harder.
But I know that we are in disagreement here. You really like subtests and I would trade them for better isolation and auto-configuration/cleanup between tests ;)

Reply:
@alexeykazakov what dependency are you talking about ? if it's about sharing the same data set between those subtests, then I have absolutely no problem with moving/duplicating the `//given` section (ie, data init) within each subtest.
About isolation, I think that Ginkgo is better at that. See the `Extracting common setup` on http://onsi.github.io/ginkgo/#structuring-your-specs

Reply:
@alexeykazakov I've duplicated the data init in each subtest in https://github.com/fabric8-services/fabric8-auth/pull/636/commits/d22a6464812250925d924622caadb142ee22b9f6